### PR TITLE
Add --with-libffi-source-dir feature and removed --enable-bundled-libffi option.

### PR DESCRIPTION
**Pull request by hsbt**
https://bugs.ruby-lang.org/issues/18571

### CODE REVIEW

**Comment by hsbt** [comment]
Note: https://github.com/ruby/fiddle/actions/runs/3202406059/jobs/5231356410 is failing with memory leak. I'm not sure why it's happend.

**Comment by kou** [comment]
> Note: https://github.com/ruby/fiddle/actions/runs/3202406059/jobs/5231356410 is failing with memory leak. I'm not sure why it's happend.

We can omit the test for now.

File: .github/workflows/libffi.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          ruby-version: ${{ matrix.ruby }}
+      - name: Set up libffi ${{ matrix.libffi }}
+        run: |
+          wget https://ftp.osuosl.org/pub/blfs/conglomeration/libffi/libffi-${{ matrix.libffi }}.tar.gz
+          tar xzf libffi-${{ matrix.libffi }}.tar.gz
+      - name: Compile libffi ${{ matrix.libffi }}
+        run: |
+          cd libffi-${{ matrix.libffi }}
+          ./configure --prefix=${{ matrix.libffi-prefix }}
+          make
```
Comment (kou) [inline .github/workflows/libffi.yml:29]:
```suggestion
          make -j$(nproc)
```

**Comment by kou** [comment]
Or we can just ignore the failure.

**Review by kou — APPROVED** [review]
+1

**Comment by hsbt** [comment]
Thanks!

**Pull request merged**